### PR TITLE
separated delete's for the boolean operations. Function volume inside

### PR DESCRIPTION
**Pull request by tryfon-mw**
Here s the first cherry pick. Separation of delete  in opencascade boolean operations methods between the input entities and the tool_entities.

Addition of a method that allows to keep the tagging of the physical geometries intact, namely volumes_inside.

### CODE REVIEW

File: pygmsh/opencascade/geometry.py [hunk 1]
```diff
@@ -129,33 +146,33 @@ def _boolean_operation(
...
                 legal_dim_types[dim_type],
                 ','.join(e.id for e in tool_entities),
-                'Delete;' if delete else ''
+                'Delete;' if delete2 else ''
                 ))
 
         return dim_type(id0=name, is_list=True)
 
-    def boolean_intersection(self, entities, delete=True):
+    def boolean_intersection(self, entities, delete1=True, delete2=True):
```
Comment (nschloe) [inline pygmsh/opencascade/geometry.py]: We'll have to think about the interface here. One supplies a list of entities to `boolean_intersection`, and it's not entirely clear to the user what `delete1` and `delete2` mean in the context. Perhaps `delete_first` and `delete_other` would be more appropriate.

File: pygmsh/opencascade/geometry.py [hunk 2]
```diff
@@ -86,13 +86,30 @@ def add_wedge(self, *args, **kwargs):
         self._GMSH_CODE.append(p.code)
         return p
 
+    def volumes_inside(self, ex_vol, in_vols):
+        """ Function uses the boolean operations in order to make given internal volumes to 
+            belong to external one.
+
+            This function exists in order to make possible the assignment to the physical volumes
+            and boundary surfaces.
```
Comment (nschloe) [inline pygmsh/opencascade/geometry.py]: I don't really understand what this wants to say. Internal, external? Perhaps a test would help clarifying things.

Comment (tryfon-mw) [reply pygmsh/opencascade/geometry.py]: This is too specific indeed. I have removed it. It only applies any boolean operation with delete_first=True and delete_other=False.

File: pygmsh/opencascade/geometry.py (hunk 1 above)
Comment (tryfon-mw) [reply pygmsh/opencascade/geometry.py]: Good point thanks. Fixed.

**Comment by nschloe** [comment]
LGTM. I'll update and merge.

**Pull request merged**